packages/firestore/src/lite-api/query.ts

```diff
@@ -87,56 +86,137 @@ export type QueryConstraintType =
+ * to a Firestore query.
+ */
+export abstract class AppliableConstraint {
+  /**
+   * Takes the provided {@link Query} and returns a copy of the {@link Query} with this
+   * {@link AppliableConstraint} applied.
+   */
+  abstract _apply<T>(query: Query<T>): Query<T>;
+}
+
 /**
  * A `QueryConstraint` is used to narrow the set of documents returned by a
  * Firestore query. `QueryConstraint`s are created by invoking {@link where},
- * {@link orderBy}, {@link (startAt:1)}, {@link (startAfter:1)}, {@link
- * endBefore:1}, {@link (endAt:1)}, {@link limit} or {@link limitToLast} and
+ * {@link orderBy}, {@link startAt}, {@link startAfter}, {@link
+ * endBefore}, {@link endAt}, {@link limit}, {@link limitToLast} and
  * can then be passed to {@link query} to create a new query instance that
  * also contains this `QueryConstraint`.
  */
-export abstract class QueryConstraint {
-  /** The type of this query constraints */
+export abstract class QueryConstraint extends AppliableConstraint {
+  /** The type of this query constraint */
   abstract readonly type: QueryConstraintType;
 
   /**
    * Takes the provided {@link Query} and returns a copy of the {@link Query} with this
-   * {@link QueryConstraint} applied.
+   * {@link AppliableConstraint} applied.
    */
   abstract _apply<T>(query: Query<T>): Query<T>;
 }
 
 /**
- * Creates a new immutable instance of {@link Query} that is extended to also include
- * additional query constraints.
+ * Creates a new immutable instance of {@link Query} that is extended to also
+ * include additional query constraints.
  *
- * @param query - The {@link Query} instance to use as a base for the new constraints.
+ * @param query - The {@link Query} instance to use as a base for the new
+ * constraints.
+ * @param compositeFilter - The {@link QueryCompositeFilterConstraint} to
+ * apply. Create {@link QueryCompositeFilterConstraint} using {@link and} or
+ * {@link or}.
+ * @param queryConstraints - Additional {@link QueryNonFilterConstraint}s to
+ * apply (e.g. {@link orderBy}, {@link limit}).
+ * @throws if any of the provided query constraints cannot be combined with the
+ * existing or new constraints.
+ * @internal TODO remove this internal tag with OR Query support in the server
+ */
+export function query<T>(
+  query: Query<T>,
+  compositeFilter: QueryCompositeFilterConstraint,
+  ...queryConstraints: QueryNonFilterConstraint[]
+): Query<T>;
+
+/**
+ * Creates a new immutable instance of {@link Query} that is extended to also
+ * include additional query constraints.
+ *
+ * @param query - The {@link Query} instance to use as a base for the new
+ * constraints.
  * @param queryConstraints - The list of {@link QueryConstraint}s to apply.
  * @throws if any of the provided query constraints cannot be combined with the
  * existing or new constraints.
  */
 export function query<T>(
   query: Query<T>,
   ...queryConstraints: QueryConstraint[]
+): Query<T>;
+
+export function query<T>(
```

Comment:
This is not publicly exposed, is it?

Reply:
It's not. Thanks for double checking. I also reviewed the api-report.md and index.d.ts, and only the original overload is exposed. For additional info on the topic, see https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads.

Reply:
Thanks! 🚀